src/tls.c

```diff
@@ -1112,6 +1112,9 @@ static int Hmac_UpdateFinal(Hmac* hmac, byte* digest, const byte* in,
             if (ret != 0)
                 break;
         }
+        /* call final to cleanup */
+        if (ret == 0)
+            ret = wc_HmacFinal(hmac, dummy);
```

Comment:
This needs to be tested for time changes.  Sean, can you review?

Reply:
@SparkiDev. Please let me know your thoughts for timing impact. The final is important for the CryptoCB case for cleanup

Reply:
So, introducing a final does cause timing issues.
I'm not sure what the asm code is doing but if you make the function not inline (take off the static keyword) then it looks fine again.

Reply:
Timing was impacted, so opted for a better cleanup solution using `wc_HmacFree`.